org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/JavacVariableBinding.java

```diff
@@ -202,7 +202,10 @@ public ITypeBinding getType() {
 
 	@Override
 	public int getVariableId() {
-		return variableSymbol.adr; // ?
+		// FIXME: since we are not running code generation,
+		// the variable has not been assigned an offset,
+		// so it's always -1.
+		return variableSymbol.adr;
```

Comment:
What about trying to use hashCode() here?

Reply:
I checked to see where the method is being used. It's being used in jdt.ui as part of a code assist. I think that the number is being used to figure out the order of appearance of the variables.